solution/index.js

```diff
@@ -0,0 +1,170 @@
+// Add new Task 
+function handleAddTask (event){
+    let newTask = event.target.nextElementSibling.value;
+    if(newTask === ""){
+        alert("can't add empty task");
+        return;
+    }
+    let newTaskEl = document.createElement("li");
+    newTaskEl.classList.add("task");
+    newTaskEl.innerText = newTask;
+    newTaskEl.contentEditable = false;
+    let taskList = document.getElementsByTagName("ul")[event.target.parentNode.id];
+    taskList.insertBefore(newTaskEl, taskList.childNodes[0]);
+    tasks[event.target.parentNode.dataset.action].unshift(newTask);  
+    localStorage.setItem("tasks", JSON.stringify(tasks));  //save to local storage
+    
+}
+//Edit Task
+function handleDBclick (event){
+    if(event.target.tagName !== "LI") return;
+    let task = event.target.innerText;
+    let targetArr = event.target.closest("section").dataset.action;
+    let taskI = tasks[targetArr].indexOf(task);
+    event.target.contentEditable = true;
+    event.target.focus();
+    event.target.onblur = () => {
+        if(event.target.innerText=="") { //prevent saving empty task
+            event.target.focus();
+            return;
+        }
+        let updateTask = event.target.innerText;
+        tasks[targetArr][taskI] = updateTask;
+        localStorage.setItem("tasks", JSON.stringify(tasks)); //save to local storage
+        event.target.contentEditable = false
+    }
+    
+}
+//load tasks from local storage
+function loadTasks () {
+    if(!localStorage.getItem("tasks")) //if no tasks found in local storage save empty tasks jason
+        localStorage.setItem("tasks", JSON.stringify(tasks))
+    tasks = JSON.parse(localStorage.getItem("tasks"));
+    let classI = 0;
+    let newLi;
+    for(let taskArr in tasks){ //loop over task arrays (to-do \ in progress \ done)
+        for(let i = 0; i < tasks[taskArr].length; i++){ //loop over each array
+            newLi =  document.createElement("li");
+            newLi.classList.add("task");
+            newLi.innerHTML = tasks[taskArr][i];
+            document.getElementsByTagName("ul")[classI].append(newLi);
+        }
+        classI++;
+    }
+}
+
+//handle task mouse over
+function handleMouseOver(event) {
+    if(event.target.tagName !== "LI") return; 
+    event.target.classList.add("active");
+    document.addEventListener("keydown", keyPressMove);
```

Comment:
This event can be initialized along with the li by default. Logically one has to be above an object for it to be clickable